core/src/type_info.rs

```diff
@@ -0,0 +1,304 @@
+            Schema::Atomic(Atomic::Scalar(BsonTypeName::Int)),
+            Nullability::NULLABLE
+        ),
+        MongoColMetadata::new(
+            "",
+            "".to_string(),
+            "INTERVAL_PRECISION".to_string(),
+            Schema::Atomic(Atomic::Scalar(BsonTypeName::Int)),
+            Nullability::NULLABLE
+        ),
+    ];
+}
+
+#[derive(Debug)]
+pub struct MongoTypesInfo {
+    current_type_index: isize,
+    sql_data_type: SqlDataType,
+}
+
+impl MongoTypesInfo {
+    pub fn new(sql_data_type: SqlDataType) -> MongoTypesInfo {
+        MongoTypesInfo {
+            current_type_index: 0,
+            sql_data_type,
+        }
+    }
+}
+
+impl MongoStatement for MongoTypesInfo {
+    // iterate through the list, searching for the next "valid" data type.
+    // a type is valid if its sql type matches the desired sql type, or if we are getting all types.
+    fn next(&mut self, _: Option<&MongoConnection>) -> Result<bool> {
+        loop {
+            self.current_type_index += 1;
+            if self.current_type_index > (DATA_TYPES.len() as isize)
+                || DATA_TYPES[(self.current_type_index - 1) as usize].sql_type == self.sql_data_type
+                || self.sql_data_type == SqlDataType(0)
+            {
+                break;
+            }
+        }
+        Ok(self.current_type_index <= (DATA_TYPES.len() as isize))
+    }
+
+    // Get the BSON value for the cell at the given colIndex on the current row.
+    fn get_value(&self, col_index: u16) -> Result<Option<Bson>> {
+        // 1 -> TYPE_NAME
+        // 2 -> DATA_TYPE
+        // 3 -> COLUMN_SIZE
+        // 4 -> LITERAL_PREFIX
+        // 5 -> LITERAL_SUFFIX
+        // 6 -> CREATE_PARAMS
+        // 7 -> NULLABLE
+        // 8 -> CASE_SENSITIVE
+        // 9 -> SEARCHABLE
+        // 10 -> UNSIGNED_ATTRIBUTE
+        // 11 -> FIXED_PREC_SCALE
+        // 12 -> AUTO_UNIQUE_VALUE
+        // 13 -> LOCAL_TYPE_NAME
+        // 14 -> MINIMUM_SCALE
+        // 15 -> MAXIMUM_SCALE
+        // 16 -> SQL_DATA_TYPE
+        // 17 -> SQL_DATETIME_SUB
+        // 18 -> NUM_PREC_RADIX
+        // 19 -> INTERVAL_PRECISION
+        match DATA_TYPES.get((self.current_type_index - 1) as usize) {
+            Some(type_info) => {
+                Ok(Some(match col_index {
+                    1 | 13 => Bson::String(type_info.type_name.to_string()),
+                    2 | 16 => Bson::Int32(type_info.sql_type.0 as i32),
+                    3 => match type_info.precision {
+                        Some(precision) => Bson::Int32(precision as i32),
+                        None => Bson::Null,
+                    },
+                    4 | 5 => match **type_info {
+                        BsonTypeInfo::DATE | BsonTypeInfo::STRING => Bson::String("'".to_string()),
+                        _ => Bson::Null,
+                    },
+                    6 => Bson::Null,
+                    7 => Bson::Boolean(**type_info == BsonTypeInfo::OBJECTID),
```

Comment:
Why is this the returned value?

Reply:
Is this re the logic or the value? Logic would be that object id (I think) should be non nullable in all cases. Value I used a boolean but should probably actually make it an integer (because it is possible the value could be in [0, 1, 2]).

Reply:
This is about the value. I still don't understand why we are specifically singling out ObejctId as "not nullable". In BSON aren't all types except Null/Missing/Undefined "not nullable" since "Null" is actually a type?

Reply:
@jchemburkar bumping this one!

Reply:
@mattChiaravalloti thanks for the bump :) Your point makes sense to me, thanks for the clarification. My mental model is stemming from this function being `intended for use in Data Definition Language (DDL) statements`, and relatedly, thinking of these as being the types of a `Column`  rather than of a specific value. Thus, following the logic of [MongoColMetadata nullability](https://github.com/mongodb/mongo-odbc-driver/blob/c1b68f647d2364f0a19adf39821ae2bedd283027/core/src/col_metadata.rs#L220), it would seem any of these types _could_ be nullable if they are not required, other than ObjectId which will populate a value regardless (?). 

This was the logic that led me to this, but definitely happy to change if there are issues or I'm misunderstanding anything!

Reply:
Per standup discussion, we will mark all types as nullable